hooks/pip-build-hook.sh

```diff
@@ -5,6 +5,10 @@ pipBuildPhase() {
     echo "Executing pipBuildPhase"
     runHook preBuild
 
+    if [ -z "$SETUPTOOLS_SCM_PRETEND_VERSION" ] && echo "$buildInputs" | grep -E 'setuptools-scm|hatch-vcs'; then
```

Comment:
Some suggestions:

- `grep --quiet` avoids printing the result.
- `grep --word-regexp` would avoid matching substrings of other package names.